src/tools/schema2cpp.ts

```diff
@@ -115,6 +115,10 @@ export class Schema2Cpp extends Schema2Base {
                     `  printer.add("${field}: ", entity.${field}_);`);
     });
 
+    if (fieldCount === 0) {
```

Comment:
Took me a while to understand this, possibly the names could be a little more instructive or benefit from a comment.

I think this is generating code for checking the equality of empty entities and we're using structural equality, so they're all eq, which makes sense. Except that type slicing might mean that they're actually not the same entity, and so this could be confusing for developers (i.e. if I have two entities with different [hidden] fields and I can't see the difference, but I also don't necessarily want them to override each other in a collection.)

Not sure this potential confusion should be dealt with in this CL, the code itself seems good.

Reply:
Yes, this is for the `fields_equal` helper method, which is a comparison that explicitly excludes the arcs id. `operator==` includes the ids for strict equality checking. Both of these only work for entities of exactly the same class; we don't have any support for automatic comparison of entities from different schemas.

I also haven't thought about how to implement type slicing in C++.